sql/sys_vars.cc

```diff
@@ -1232,6 +1232,12 @@ static Sys_var_mybool Sys_binlog_rows_query(
        SESSION_VAR(binlog_rows_query_log_events),
        CMD_LINE(OPT_ARG), DEFAULT(FALSE));
 
+static Sys_var_mybool Sys_binlog_ddl_query(
+       "binlog_ddl_query_log_events",
+       "Allow writing of DDl_query_log events into binary log.",
+       SESSION_VAR(binlog_ddl_query_log_events),
+       CMD_LINE(OPT_ARG), DEFAULT(FALSE));
+
```

Comment:
1. Why stripping of comments is a problem? (from the user perspective)?
2. Why do we need a new option? Will the presence of comments create any backward compatibility issues?
3. Why fix only for DROP TABLE? what about ALTER, CREATE?
4. Aren't DMLs affected?

Reply:
Please find below answers for questions : 
1. Stripping of comments is problem because Customer's application depends on metadata they include into queries.
2. We need option so that only subset of customers , who wish to change default behaviour of drop table could enable option , for rest of customers's no change , since this option is disabled by default, so no compatibility issue.
3. Fix for drop table only , because drop table only prepares queries to be logged in bin log dynamically , I will check again this for ALTER table. For create table there is no problem.
4. This is only for binlog logging.

Reply:
I don't think this change warrants a new option. It just that, after the fix, comments are logged for DROP TABLE. Does it break any compatibility?

Unless, we are changing query behaviour (output change), disk format changes, we don't need an option.

If there are no compatibility issues, I suggest to get rid of the option.

Reply:
also let me know your findings on ALTER TABLE comments

Reply:
ALTER TABLE doesn't strip comments , logs statements as they are entered on mysql client.
New option is added to keep default behaviour of drop commands , Only in exceptional cases like 1 customer request , we are just providing a way to get away from default behaviour and keep comments to be bin-logged .
By default option is disabled , so no compatibility problem for anyone.

Reply:
I understood the behaviour/impact of the option. My question is, if it is enabled by default, is there any problem for other users (not customer)? Will it break any application/format for others?